pkg/router/template/template_helper.go

```diff
@@ -80,7 +105,7 @@ func matchValues(s string, allowedValues ...string) bool {
 
 func matchPattern(pattern, s string) bool {
 	log.V(7).Info("matchPattern called", "pattern", pattern, "s", s)
-	status, err := regexp.MatchString(`\A(?:`+pattern+`)\z`, s)
+	status, err := matchString(`\A(?:`+pattern+`)\z`, s)
```

Comment:
`matchPattern` isn't used in the template, so this shouldn't affect performance (but it shouldn't hurt anything either).

Reply:
This was explicit as `MatchString` compiles a regexp too:

```go
// MatchString reports whether the string s
// contains any match of the regular expression pattern.
// More complicated queries need to use Compile and the full Regexp interface.
func MatchString(pattern string, s string) (matched bool, err error) {
	re, err := Compile(pattern)
	if err != nil {
		return false, err
	}
	return re.MatchString(s), nil
}
```

Reply:
I mean, this entire function is effectively dead code anyway, unless someone is using a custom template that uses `matchPattern`.  That said, it is possible that someone _is_ using a custom template that uses `matchPattern`, so we might as well optimize it.